### PR TITLE
INT-175 slim IAM permissions

### PR DESCRIPTION
**Pull request by mburns**
- [x] create custom role
- [x] add `dataproc.batches.submit.pyspark` permission
- [x] remove `roles/bigquery.readSessionUser` as redundant

TODO:
- [ ] decide on `dataproc.batches` perms
- [ ] consolidate/exercise BigQuery perms

### CODE REVIEW

**Comment by mburns** [comment]
Currently, the CI test scripts don't exercise the BigQuery features, so `bigquery.dataEditor` and `bigquery.user` roles aren't strictly required.

File: preflight.sh [hunk 1]
```diff
@@ -40,17 +37,13 @@ gcloud projects add-iam-policy-binding "$PROJECT_ID" \
 	--member="serviceAccount:$SA_MAIL" \
 	--role=roles/bigquery.dataEditor
 
-gcloud projects add-iam-policy-binding "$PROJECT_ID" \
-	--member="serviceAccount:$SA_MAIL" \
-	--role=roles/bigquery.readSessionUser
```
Comment (mburns) [inline preflight.sh:45]: this is a strict subset of `roles/bigquery.user`.

File: gcp_permissions.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+title: "Integral Engine"
+description: Integral Engine required permissions for Gabor
+stage: ALPHA
+includedPermissions:
+  # - dataproc.batches.create
```
Comment (mburns) [inline gcp_permissions.yml:5]: https://cloud.google.com/dataproc-serverless/docs/concepts/iam

File: .github/workflows/cron_test_deploy.yml [hunk 3]
```diff
@@ -16,6 +16,7 @@ jobs:
       SA_NAME: test-sa-${{ github.run_id }}-${{ github.run_attempt }}
       NET_NAME: integral-engine-${{ github.run_id }}-${{ github.run_attempt }}
       DEP_BUCKET: integral-engine-bucket-${{ github.run_id }}-${{ github.run_attempt }}
+      ROLE_NAME: integral.dataproc.worker_${{ github.run_id }}_${{ github.run_attempt }}
```
Comment (johnlk) [inline .github/workflows/cron_test_deploy.yml:19]: might have to rebase off the PR I just approved

File: gcp_permissions.yml (hunk 2 above)
Comment (johnlk) [reply gcp_permissions.yml:5]: It's possible even the impersonating account needs some dataproc perms like the batches.create. I say this from my experience trying to run PL's test locally

Comment (johnlk) [reply gcp_permissions.yml:5]: I guess my thought would be to exclude it unless we KNEW we needed it, and this cron test will let us know

**Review by johnlk — APPROVED** [review]
LGTM

**Pull request merged**